### PR TITLE
feat(config): watch config and font features

### PR DESCRIPTION
**Pull request by TwIStOy**
After some discussion in #1920 

This pr includes

- Watch `config.toml`. Only `font` can be hot-reloaded now.
- Font settings now looks like [`alacritty`](https://alacritty.org/config-alacritty.html).
- Support `font.features` settings

## What kind of change does this PR introduce?
- Feature

## Did this PR introduce a breaking change? 
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
- No

### CODE REVIEW

**Comment by fredizzimo** [comment]
Thank you very much for taking time to implement this!

I don’t have acess to a computer to run Neovide at the moment, so I can’t really test it and give feedback at the moment.

But I think someone else will take a look shortly. Maybe some documentation would be good to add, so we can better see if we want to make some small tweaks to the schema.

File: src/settings/config.rs [hunk 1]
```diff
@@ -42,16 +51,69 @@ pub struct Config {
...
+        let mut watcher = notify::RecommendedWatcher::new(
+            tx,
+            notify::Config::default().with_compare_contents(true),
+        )
+        .unwrap();
+        watcher
+            .watch(&config_path(), notify::RecursiveMode::NonRecursive)
+            .unwrap();
+        std::mem::forget(watcher);
+        std::thread::spawn(move || {
```
Comment (MultisampledNight) [inline src/settings/config.rs:85]: Might fit better as a standalone function in order to avoid too much nesting.

File: src/settings/config.rs [hunk 2]
```diff
@@ -42,16 +51,69 @@ pub struct Config {
...
+            let mut previous_config = init_config;
+            loop {
+                match rx.recv() {
+                    Ok(_) => {
```
Comment (MultisampledNight) [inline src/settings/config.rs:89]: Could make use of early return.

**Review by MultisampledNight — COMMENTED** [review]
Thank you _a lot_, for the past and this PR! That's a lot of work, whew. I'll take care of what I annotated and then merge this!

**Comment by MultisampledNight** [comment]
Chooooooooooo chooooooooooo! Thank you! :steam_locomotive:

**Pull request merged**